pytools/test/test_persistent_dict.py

```diff
@@ -479,6 +479,19 @@ def test_scalar_hashing() -> None:
         assert keyb(np.complex256(1.1+2.2j)) == keyb(np.complex256(1.1+2.2j))
     assert keyb(np.clongdouble(1.1+2.2j)) == keyb(np.clongdouble(1.1+2.2j))
 
+    h_int64 = keyb(np.int64(1))
+
+    import unittest.mock
+    with unittest.mock.patch("pytools.persistent_dict.IS_BIGENDIAN", True):
+        # Very crude big-endian test. Apparently, there is no way to
+        # create a numpy scalar (not an np.array!) of a specific byte order.
+        if sys.byteorder == "big":
+            # no need to swap
+            be_val = np.int64(1)
+        else:
+            be_val = np.int64(1).byteswap()
```

Comment:
This part is kinda redundant on big-endian systems, so I'm wondering: how about testing little endian there instead? i.e. something like:

```suggestion
    with unittest.mock.patch("pytools.persistent_dict.IS_BIGENDIAN", sys.byteorder == "little"):
        # Very crude test that the other endianness works as well.
        be_val = np.int64(1).byteswap()
```

Reply:
Thanks, what do you think of [4ce5d88](https://github.com/inducer/pytools/pull/285/commits/4ce5d88b649b3663b303255047ff9fe408027c46)?

Out of curiosity, what is the PPC64 hardware that you are using?

Reply:
Yep, that matches my idea precisely and passes on our machine.

To be honest, I don't know exactly, I'm only an user there — apparently it's some kind of virtualized POWER9 server.

Reply:
> Yep, that matches my idea precisely and passes on our machine.
> 
> To be honest, I don't know exactly, I'm only an user there — apparently it's some kind of virtualized POWER9 server.

Ah interesting, I was under the impression that many Power9s were running in LE mode.

Reply:
Yeah, PPC64 is bi-endian. Since Gentoo still supports big-endian PPC64, we literally have two testing machines: one set to run LE, the other BE. In fact, it's quite a nice setup, because we can test packages on two very similar systems with different endianness, and so easily distinguish endianness-related issues from other kinds of problems that occur on BE hardware such as SPARC.